src/Text.Analyzers/Core (Net4)/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -0,0 +1,416 @@
+                    AnalyzeIdentifiers(syntaxTreeContext);
+                });
+            });
+        }
+
+        /// <summary>
+        /// Check all IdentifierTokens for names that are either too short or misspelled.
+        /// </summary>
+        /// <param name="context">Object containing information about the syntax tree.</param>
+        private void AnalyzeIdentifiers(SyntaxTreeAnalysisContext context)
+        {
+            foreach (var token in context.Tree.GetRoot().DescendantTokens().Where(x => x.IsKind(SyntaxKind.IdentifierToken)))
+            {
+                var identifier = token.ValueText;
+                if ((token.Parent is TypeParameterSyntax && identifier.StartsWith("T", StringComparison.InvariantCulture)) ||
+                    (token.Parent is InterfaceDeclarationSyntax && identifier.StartsWith("I", StringComparison.InvariantCulture)))
+                {
+                    // Interfaces and type parameters are expected to start with T and I, respectively. Thus, we need to
+                    // strip those characters to know if this identifier is meaningful
+                    identifier = identifier.Substring(1);
+                }
+
+                // Any identifier with more than one character is considered meaningful
+                if (identifier.Length == 1)
+                {
+                    var diagnostic = GetUnmeaningfulIdentifierDiagnostic(token);
+                    if (diagnostic != null)
+                    {
+                        context.ReportDiagnostic(diagnostic);
+                        continue;   // no reason to check this identifier for misspellings
+                    }
+                }
+
+                foreach (var misspelledWord in GetMisspelledWords(token.ValueText))
+                {
+                    var diagnostic = GetMisspelledWordDiagnostic(token, misspelledWord);
+                    if (diagnostic != null)
+                    {
+                        context.ReportDiagnostic(diagnostic);
+                    }
+                }
+            }
+        }
+
+        private IEnumerable<string> GetMisspelledWords(string name)
+        {
+            if (!_namingService.SupportsSpelling)
+            {
+                yield break;
+            }
+
+            var parser = new WordParser(name, WordParserOptions.SplitCompoundWords);
+            if (parser.PeekWord() != null)
+            {
+                var word = parser.NextWord();
+
+                do
+                {
+                    if (_namingService.CheckSpelling(word) != WordSpelling.Unrecognized)
+                    {
+                        continue;
+                    }
+
+                    yield return word;
+                }
+                while ((word = parser.NextWord()) != null);
+            }
+        }
+
+        private Diagnostic GetUnmeaningfulIdentifierDiagnostic(SyntaxToken token)
+        {
+            return GetDiagnostic(
```

Comment:
:memo: The delegates here need to be rewritten to avoid allocating every time this method is called

Reply:
No longer applies